File: app/src/main/java/com/tutpro/baresip/History.java
package com.tutpro.baresip;

import java.io.Serializable;
import java.util.GregorianCalendar;

public class History implements Serializable {

    private static final long serialVersionUID = -299482035708790407L;

    private String ua, call, aor, peer_uri, direction;
    private GregorianCalendar time;
    private Boolean connected;

    public History(String ua, String call, String aor, String peer_uri, String direction,
                   Boolean connected) {
        this.ua = ua;
        this.call = call;
        this.aor = aor;
        this.peer_uri = peer_uri;
        this.direction = direction;
        this.time = new GregorianCalendar();
        this.connected = connected;
    }

    public String getUA() { return ua; }
    public String getCall() { return call; }
    public String getAoR() { return aor; }
    public String getPeerURI() { return peer_uri; }
    public String getDirection() { return direction; }
    public GregorianCalendar getTime() { return time; }
    public Boolean getConnected() { return connected; }
}

File: app/src/main/kotlin/com/tutpro/baresip/History.kt
package com.tutpro.baresip

import java.io.Serializable
import java.util.GregorianCalendar

class History(val aor: String, val peerURI: String, val direction: String,
              val connected: Boolean) : Serializable {

    val time: GregorianCalendar = GregorianCalendar()

    companion object {

        fun aorHistory(history: ArrayList<History>, aor: String): Int {
            var size = 0;
            for (h in history) {
                if (h.aor == aor) size++
            }
            return size
        }

        fun aorRemoveHistory(history: ArrayList<History>, aor: String) {
            for (h in history) {
                if (h.aor == aor) {
                    history.remove(h)
                    return
                }
            }
        }

        fun aorLatestHistory(history: ArrayList<History>, aor: String): History? {
            for (h in history.reversed())
                if (h.aor == aor) return h
            return null
        }

    }


}
